File: library/src/main/java/com/alamkanak/weekview/EventLongPressListener.java
package com.alamkanak.weekview;

import android.graphics.RectF;
import android.support.annotation.NonNull;

public interface EventLongPressListener<T> {

    /**
     * Similar to {@link EventClickListener} but with a long press.
     *
     * @param data:     event clicked.
     * @param eventRect: view containing the clicked event.
     */
    void onEventLongPress(@NonNull T data, @NonNull RectF eventRect);

}

File: library/src/main/java/com/alamkanak/weekview/EventLongPressListener.kt
package com.alamkanak.weekview

import android.graphics.RectF

@FunctionalInterface
interface EventLongPressListener<T> {

    /**
     * Called when an [EventChip] is long-clicked.
     *
     * @param data The [T] object associated with the [EventChip]'s [WeekViewEvent]
     * @param eventRect The [RectF] of the [EventChip]
     */
    fun onEventLongPress(data: T, eventRect: RectF)

}
